### PR TITLE
Feature/error handling

### LINKED ISSUE #167: Proper API error responses
Implement properly formatted errors and error codes. Map exceptions to HTTP errors with problem details. For every error we require at least one integration test

### PR DESCRIPTION
**Pull request by yorickdewid**
**What issue does this PR address?**
Fixes #167 

**Does this PR introduce a breaking change?**
Nope

### CODE REVIEW

**Comment by yorickdewid** [comment]
Wait on #288

File: src/FunderMaps.Webservice/Startup.cs [hunk 1]
```diff
@@ -80,6 +80,7 @@ public void ConfigureServices(IServiceCollection services)
             {
                 // FUTURE: The full enum description support for swagger with System.Text.Json is a WIP. This is a custom tempfix.
                 c.SchemaFilter<EnumSchemaFilter>();
+                // FUTURE: This call is obsolete.
```
Comment (tabeckers) [inline src/FunderMaps.Webservice/Startup.cs:83]: Hmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm

File: tests/FunderMaps.IntegrationTests/Backend/Application/AuthTests.cs [hunk 2]
```diff
@@ -69,6 +69,36 @@ public async Task RefreshSignInReturnSuccessAndToken()
...
+                .CreateClient();
+
+            // Act
+            var response = await client.PostAsJsonAsync("api/auth/signin", signIn);
+            var returnObject = await response.Content.ReadFromJsonAsync<ProblemModel>();
+
+            // Assert
+            Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+            Assert.Equal((short)HttpStatusCode.Unauthorized, returnObject.Status);
+            Assert.Contains("Login", returnObject.Title);
```
Comment (tabeckers) [inline tests/FunderMaps.IntegrationTests/Backend/Application/AuthTests.cs:99]: This seems a bit too vague for a testcase. Either check the full message or don't check it, but let's change that in the future. Fine for now.

**Review by tabeckers — CHANGES_REQUESTED** [review]
Made some comments, one change request and one question.

File: src/FunderMaps.AspNetCore/FunderMapsExceptionMapper.cs [hunk 3]
```diff
@@ -20,8 +19,15 @@ public sealed class FunderMapsExceptionMapper : ExceptionMapperBase<FunderMapsCo
...
             {
-                ProductNotFoundException x => BuildMessage("Could not find product", HttpStatusCode.Conflict),
-                _ => BuildMessage("Unable to process request", HttpStatusCode.InternalServerError)
+                AuthenticationException _ => BuildMessage("Login attempt failed with provided credentials.", HttpStatusCode.Unauthorized),
+                AuthorizationException _ => BuildMessage("Access to resource forbidden.", HttpStatusCode.Forbidden),
+                EntityNotFoundException _ => BuildMessage("Requested entity not found.", HttpStatusCode.NotFound),
+                EntityReadOnlyException _ => BuildMessage("Requested entity is immutable.", HttpStatusCode.Locked),
+                InvalidCredentialException _ => BuildMessage("Action failed with provided credentials.", HttpStatusCode.Forbidden),
+                InvalidProductRequestException _ => BuildMessage("Invalid product requested.", HttpStatusCode.BadRequest),
+                ProductNotFoundException _ => BuildMessage("Requested product not found.", HttpStatusCode.Forbidden),
```
[the remark this answers is not in the document]
Comment (yorickdewid) [reply src/FunderMaps.AspNetCore/FunderMapsExceptionMapper.cs:28]: Moet je InvalidProductRequestException gebruiken

File: src/FunderMaps.WebApi/Controllers/Application/AuthController.cs [hunk 4]
```diff
@@ -24,6 +24,11 @@ public class AuthController : BaseApiController
         public AuthController(AuthenticationHelper authenticationHelper) 
             => _authenticationHelper = authenticationHelper ?? throw new ArgumentNullException(nameof(authenticationHelper));
 
+        /// <summary>
```
[the remark this answers is not in the document]
Comment (yorickdewid) [reply src/FunderMaps.WebApi/Controllers/Application/AuthController.cs:27]: :angry:

File: src/FunderMaps.Core/Exceptions/RepositoryException.cs [hunk 5]
```diff
@@ -15,7 +15,7 @@ namespace FunderMaps.Core.Exceptions
     ///         specifc derived exception fits the case.
     ///     </para>
     /// </remarks>
-    public class RepositoryException : FunderMapsCoreException
+    public abstract class RepositoryException : FunderMapsCoreException
```
[the remark this answers is not in the document]
Comment (yorickdewid) [reply src/FunderMaps.Core/Exceptions/RepositoryException.cs:18]: We may be able to remove this exception alltogether

File: src/FunderMaps.Webservice/Startup.cs (hunk 1 above)
Comment (yorickdewid) [reply src/FunderMaps.Webservice/Startup.cs:83]: Hmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm

File: src/FunderMaps.Core/Exceptions/RepositoryException.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (yorickdewid) [reply src/FunderMaps.Core/Exceptions/RepositoryException.cs:18]: Done

**Pull request merged**